### DISCUSSION TITLE
Is there a way to publish from a different branch on a username.github.io repository?

### DISCUSSION THREAD

**Discussion by geocine** [community/community#23029] [Other Feature Feedback, Questions, & Ideas]
<p>Right now this is how GitHub Pages work:</p>
<ul>
<li>If the name of the repository is <strong><a href="http://username.github.io" rel="nofollow noopener">username.github.io</a></strong> , it pulls from the <strong>master</strong> branch.</li>
<li>If it’s anything else, it pulls from the <strong>gh-pages</strong> branch or any branch that you configure</li>
</ul>
<p>Is there a way to publish from a different branch on a <strong><a href="http://username.github.io" rel="nofollow noopener">username.github.io</a></strong> repository?</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hey <a class="mention" href="/geocine">@geocine</a>,</p>
<p>Thanks or being here! AFAIK GitHub Pages only lets you configure one publishing source for a particular repository. </p>

**Comment by geocine** [comment]
<p>Thanks <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> , do you know if there are any future plans to change this . Allow to choose a branch similar to other repositories?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Not at this time, although we do have an internal issue discussing this, I’ll be sure to add your feedback to it.</p>
